src/NumericalAlgorithms/Interpolation/IrregularInterpolant.cpp

```diff
@@ -4,15 +4,41 @@
 #include "IrregularInterpolant.hpp"
 
 #include <array>
+#include <vector>
 
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
 #include "NumericalAlgorithms/Spectral/Mesh.hpp"
 #include "NumericalAlgorithms/Spectral/Spectral.hpp"
 // IWYU pragma: no_forward_declare Tensor
 
 namespace {
 
+// Just linear for now, can be extended to higher order...
+// Future optimization?  use Blaze sparse matrices
+std::vector<double> fd_stencil(const std::vector<double>& xi_source,
```

Comment:
why not take `DataVector` here instead of `std::vector<double>`? I realize you need to copy in eta and zeta dirs anyway, but you could save allocs and do non-owning in the xi direction